core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -443,9 +447,26 @@ default void setResponseTimeout(Duration responseTimeout) {
      * });
      * ...
      * }</pre>
+     *
+     * @deprecated Use {@link #whenResponseTimingOut()} or {@link #whenResponseTimedOut()}
      */
+    @Deprecated
     void setResponseTimeoutHandler(Runnable responseTimeoutHandler);
 
+    /**
+     * Returns a {@link CompletableFuture} which is completed when {@link ClientRequestContext} is about to
+     * get timed out.
+     */
+    CompletableFuture<Void> whenResponseTimingOut();
+
+    /**
+     * Returns a {@link CompletableFuture} which is completed when {@link ClientRequestContext} has been
```

Comment:
when {@link ClientRequestContext} has been -> after {@link ClientRequestContext} is
to clarify the difference with `whenResponseTimingOut`?

Reply:
That makes sense.